src/utils.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect, useCallback } from 'react';
+
+// custom react hook to enable and connect users Marina account.
+function useChecks() {
+  const [isInstalled, setIsInstalled] = useState(false);
+  const [isConnected, setIsConnected] = useState(false);
+
+  const isMarinaInstalled = async () => {
+    if (window.marina) {
+      setIsInstalled(!isInstalled);
+    } else {
+      setIsInstalled(isInstalled);
+    }
+  };
+
+  const checkEnabled = useCallback(async () => {
+    let res = await window.marina.isEnabled();
+    if (res) {
+      setIsConnected(!isConnected);
+    } else {
+        document
+        .getElementById('btn1')
+        .addEventListener('click', async () => await window.marina.enable());
+      document
+        .getElementById('btn2')
+        .addEventListener('click', async () => await window.marina.enable());
+      setIsConnected(!isConnected);
+    }
+  }, []);
+
+  useEffect(() => {
+    isMarinaInstalled();
+    checkEnabled();
```

Comment:
This cannot works. If marina is not installed, checking if is is enabled will crash the application.

Reply:
fixed this